src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3582,24 +3552,33 @@ void EmitSingleCharAtomicLoop(RegexNode node, bool emitLengthChecksIfRequired =
                 {
                     TransferSliceStaticPosToPos(); // we don't use static position for rtl
 
-                    string expr = $"inputSpan[pos - {iterationLocal} - 1]";
-                    if (node.IsSetFamily)
+                    if (node.IsSetFamily && maxIterations == int.MaxValue && node.Str == RegexCharClass.AnyClass)
                     {
-                        expr = MatchCharacterClass(options, expr, node.Str!, negate: false, additionalDeclarations, requiredHelpers);
+                        // If this loop will consume the remainder of the input, just set the iteration variable
```

Comment:
Should this be checking for Multiline/Singleline option in order to know if we should consume new lines as well?

Reply:
It doesn't need to.  Multiline only affects how anchors are treated, and Singleline just serves to change `.` to mean AnyClass, which we're checking for in the if clause.